[iso9660] Index out of range panic

I'm getting this panic as part of running the [linuxkit metadata service](https://github.com/linuxkit/linuxkit/tree/master/pkg/metadata)

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/linuxkit/linuxkit/pkg/metadata/vendor/github.com/diskfs/go-diskfs/filesystem/iso9660.parseDirEntry(0xa60458, 0x0, 0x0, 0xc0000e4000, 0x17000, 0x0, 0x0)
	/go/src/github.com/linuxkit/linuxkit/pkg/metadata/vendor/github.com/diskfs/go-diskfs/filesystem/iso9660/directoryentry.go:281 +0x789
github.com/linuxkit/linuxkit/pkg/metadata/vendor/github.com/diskfs/go-diskfs/filesystem/iso9660.Read(0x818500, 0xc00000e068, 0x400000, 0x0, 0x1000, 0x1, 0x8122e0, 0xc000010ed0)
	/go/src/github.com/linuxkit/linuxkit/pkg/metadata/vendor/github.com/diskfs/go-diskfs/filesystem/iso9660/iso9660.go:223 +0x965
github.com/linuxkit/linuxkit/pkg/metadata/vendor/github.com/diskfs/go-diskfs/disk.(*Disk).GetFilesystem(0xc000078230, 0x0, 0x0, 0xc000078230, 0x0, 0x0)
	/go/src/github.com/linuxkit/linuxkit/pkg/metadata/vendor/github.com/diskfs/go-diskfs/disk/disk.go:227 +0x241
```

when i mount the cdrom in linux it works fine and i can list the files on it and view them

here is an image of the cdrom: [broken-cdrom.img.gz](https://github.com/diskfs/go-diskfs/files/6240080/broken-cdrom.img.gz)

--- a/filesystem/iso9660/directoryentry.go
+++ b/filesystem/iso9660/directoryentry.go
@@ -2,6 +2,7 @@ package iso9660
 
 import (
 	"encoding/binary"
+	"errors"
 	"fmt"
 	"os"
 	"path"
@@ -278,6 +279,9 @@ func dirEntryFromBytes(b []byte, ext []suspExtension) (*directoryEntry, error) {
 // and parses it, including pulling in continuation entry bytes
 func parseDirEntry(b []byte, f *FileSystem) (*directoryEntry, error) {
 	// empty entry means nothing more to read - this might not actually be accurate, but work with it for now
+	if len(b) < 1 {
+		return nil, errors.New("cannot parse zero length directory entry")
+	}
 	entryLen := int(b[0])
 	if entryLen == 0 {
 		return nil, nil

--- a/filesystem/iso9660/iso9660.go
+++ b/filesystem/iso9660/iso9660.go
@@ -223,6 +223,9 @@ func Read(file util.File, size int64, start int64, blocksize int64) (*FileSystem
 	if read != len(b) {
 		return nil, fmt.Errorf("Root directory entry size, read %d bytes instead of expected %d", read, len(b))
 	}
+	if b[0] == 0 {
+		return nil, fmt.Errorf("Root directory entry size at location %d was zero, check header and blocksize, given as %d", location, blocksize)
+	}
 	// now read the whole entry
 	b = make([]byte, b[0])
 	read, err = file.ReadAt(b, location)

